lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs = [], isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
```

Comment:
This is an important change, minimist was eating all the arguments given in the command line, that meant that all the arguments possible for commands like [yarn install --production](https://classic.yarnpkg.com/en/docs/cli/install) and [yarn publish --access=public](https://classic.yarnpkg.com/en/docs/cli/publish) (just to give some examples), were impossible to use with **narn**

Reply:
That's why for a couple of cases It's important to just pass those arguments

Reply:
Not all flags to yarn can also be applied to npm and pnpm. For example, `yarn install --frozen-lockfile` should be converted to `npm ci`, not to `npm install --frozen-lockfile`.

We should have minimist capture the flags we want to handle and process them individually

Reply:
well, one thing is capturing and transforming when necessary, but another thing is just letting minimist just eat all the arguments and not pass them around, I didn't remove the chance of doing that transformation, I just gave it the possibility of passing arguments, before doing this it was impossible, I even [added a simple test](https://github.com/joeldenning/narn/pull/13/files?file-filters%5B%5D=.js#diff-5469b2d37b0f77aa3da967d9a6102968) to prove this bad behaviour

Reply:
And if one specific command doesn't recognize a specific argument it doesn't break or anything, it just won't do anything, it's not a breaking change at all.

For example:
yarn list --whatever
npm list --whatever
pnpm list --whatever

the fact that an argument called "whatever" exists, and it passed, and it's not recognized, doesn't break the command itself.

Reply:
Before your change, narn never did the wrong thing. After your change, narn does do the wrong thing sometimes.

We should support it correctly or not at all